### DISCUSSION TITLE
Is crane cache-friendly?

### DISCUSSION THREAD

**Discussion by bbigras** [ipetkov/crane#311] [Q&A] [answered]
I got this huge deps. It gets deduplicated on the server side, but it's a bit of a pain to upload those (I guess every time any dep changes).

![image](https://user-images.githubusercontent.com/24027/235728139-73cfc18f-d4a9-4e27-8187-f89fd003a04c.png)

**Comment by dpc** [comment, marked as answer]
`crane` will build your dependencies as a separate derivation with minimized source code, so if your `Cargo.toml` files (your dependencies) don't change, you won't have to re-build and re-upload them. If they do, you will have a rebuild and a new derivation artifact created.

**Reply by bbigras** [reply]
No way to split that into 1 derivation per dependency?

**Reply by dpc** [reply]
No. I'd say the big point of crane is avoiding the complexity of splitting derivations. If you need per-crate derivation check crate2nix , but just be aware that it might have limitations and extra complexity: https://github.com/kolloch/crate2nix#known-restrictions

**Reply by ipetkov** [reply]
If you really wanted to put in the effort to build the entire dependency tree one crate at a time (via `cargo build -p whatever`) you _could_ though it will still result in the same total size of artifacts that need to be copied over to the builder (and similarly pushed to the cache) even after dealing with the annoyance of setting up the derivations that way.

If you only care about building a subset of your workspace you can always try building only specific crates or thinning out your dependency tree
